src/compiler/checker.ts

```diff
@@ -5603,18 +5603,29 @@ namespace ts {
             return compareTypes(getTypeOfSymbol(sourceProp), getTypeOfSymbol(targetProp));
         }
 
+        // A source signature matches a target signature if the two signatures have the same number of required,
+        // optional, and rest parameters.
+        function isMatchingSignature(source: Signature, target: Signature) {

```

Comment:
Not sure "arity" conveys it either. It's about more than the number of parameters. I'm inclined to leave it as is unless there's a better suggestion.